lib/charms/data_platform_libs/v0/database_requires.py

```diff
@@ -0,0 +1,267 @@
+# Copyright 2022 Canonical Ltd.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Relation requirer side abstraction for database relation.
+
+This library is mostly an uniform interface to a selection of common databases
+metadata, with added custom events that add convenience to database management,
+and methods to consume the application related data.
+
+Following an example of using the DatabaseCreatedEvent, in the context of the
+application charm code:
+
+```python
+
+from charms.data_platform_libs.v0.database_requires import DatabaseRequires
+
+class ApplicationCharm(CharmBase):
+    # Application charm that connects to database charms.
+
+    def __init__(self, *args):
+        super().__init__(*args)
+
+        # Charm events defined in the database requires charm library.
+        self.database = DatabaseRequires(self, "database")
```

Comment:
Just to be super explicit for documentation's sake, I'd specify this as a keyword arg:

```suggestion
        self.database = DatabaseRequires(self, relation_name="database")
```

Reply:
Nice! Updated on https://github.com/canonical/data-platform-libs/pull/1/commits/ba58e91a5c7b2822d84e60089c318f25774c6ec7.